packages/lab/src/stepped-tracker/SteppedTracker.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+  ComponentPropsWithoutRef,
+  forwardRef,
+  ReactElement,
+  ReactNode,
+  Children,
+  useState,
+  isValidElement,
+  useEffect,
+  createContext,
+  useContext,
+  useMemo,
+} from "react";
+import { clsx } from "clsx";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import { makePrefixer } from "@salt-ds/core";
+
+import steppedTrackerCss from "./SteppedTracker.css";
+
+import useDetectTruncatedText, {
+  GetOverflowRef,
+} from "./useDetectTruncatedText";
+
+export type SteppedTrackerContextType = {
+  activeStep: number;
+  getOverflowRef: GetOverflowRef;
+  hasTooltips: boolean;
+  totalSteps: number;
+  isWithinSteppedTracker: boolean;
+};
+
+const defaultSteppedTrackerContext = {
+  activeStep: 0,
+  hasTooltips: false,
+  totalSteps: 1,
+  getOverflowRef: () => undefined,
+  isWithinSteppedTracker: false,
+};
+
+const SteppedTrackerContext = createContext(
+  defaultSteppedTrackerContext as unknown as SteppedTrackerContextType
+);
+
+export const useSteppedTrackerContext = () => useContext(SteppedTrackerContext);
+
+type TrackerStepNumberContextType = number;
+
+const TrackerStepContext = createContext<TrackerStepNumberContextType>(0);
+
+export const useTrackerStepContext = () => useContext(TrackerStepContext);
```

Comment:
can you please separate the context into it's own file

Reply:
Any particular reason?